### PR TITLE
Vgoncharova/mvc

### PR DESCRIPTION
**Pull request by VGoncharova**
Wishlist application for adding and removing wishes to Wishlist that uses FreeMarker templates and Exposed

### CODE REVIEW

File: mvc-web/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    mavenCentral()
+    maven { url = uri("https://maven.pkg.jetbrains.space/public/p/ktor/eap") }
+}
+
+dependencies {
+    implementation("io.ktor:ktor-server-core-jvm")
+    implementation("org.jetbrains.exposed:exposed-core:$exposed_version")
+    implementation("org.jetbrains.exposed:exposed-dao:$exposed_version")
+    implementation("org.jetbrains.exposed:exposed-jdbc:$exposed_version")
+    implementation("com.h2database:h2:2.1.214")
```
Comment (e5l) [inline mvc-web/build.gradle.kts:31]: Could you please extract the version in the properties file?

File: mvc-web/src/main/kotlin/com/example/model/DatabaseFactory.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import com.example.model.dao.wishlist.Wishlists
+import kotlinx.coroutines.Dispatchers
+import org.jetbrains.exposed.sql.Database
+import org.jetbrains.exposed.sql.SchemaUtils
+import org.jetbrains.exposed.sql.transactions.experimental.newSuspendedTransaction
+import org.jetbrains.exposed.sql.transactions.transaction
+
+object DatabaseFactory {
+    fun init() {
+        val driverClassName = "org.h2.Driver"
```
Comment (e5l) [inline mvc-web/src/main/kotlin/com/example/model/DatabaseFactory.kt:12]: Please move the DB configuration keys to the `application.yaml` or `application.properties`

**Review by e5l — CHANGES_REQUESTED** [review]
Hey @VGoncharova, good job!

Could you please address the comments?


@vnikolova, please also take a look

File: mvc-web/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+## Running
+
+Execute this command to run the sample:
+
+```bash
+./gradlew run
+```
+
+Then, navigate to [http://localhost:8080/](http://localhost:8080/) to see the sample home page.localhost:8080/](http://localhost:8080/) in a browser to create, edit, and delete articles.
```
Comment (vnikolova) [inline mvc-web/README.md:13]: There's nothing at http://localhost:8080/. This line should say:

"Then, navigate to the following pages:
- [http://localhost:8080/wish/list](http://localhost:8080/wish/list) to view and modify the wishlist. 
- [http://localhost:8080/wish/topwishes](http://localhost:8080/wish/topwishes) to see the top three wishes in the list."

File: mvc-web/src/main/kotlin/com/example/model/DatabaseFactory.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package com.example.model
+
+import com.example.model.dao.wishlist.Wishlists
+import kotlinx.coroutines.Dispatchers
+import org.jetbrains.exposed.sql.Database
+import org.jetbrains.exposed.sql.SchemaUtils
+import org.jetbrains.exposed.sql.transactions.experimental.newSuspendedTransaction
+import org.jetbrains.exposed.sql.transactions.transaction
+
+object DatabaseFactory {
```
Comment (vnikolova) [inline mvc-web/src/main/kotlin/com/example/model/DatabaseFactory.kt:10]: In other examples, we renamed "DatabaseFactory" to "DatabaseSingleton". I think we should keep the names consistent.

File: mvc-web/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# MVC application for Ktor
```
Comment (vnikolova) [inline mvc-web/README.md:1]: "MVC application with Ktor Server"

File: mvc-web/src/test/kotlin/com/example/ApplicationTest.kt [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import io.ktor.server.testing.*
+import kotlin.test.*
+
+class ApplicationTest {
+    @Test
+    fun testRoot() = testApplication {
+        application {
+            configureRouting()
+        }
+        client.get("/").apply {
```
Comment (vnikolova) [inline mvc-web/src/test/kotlin/com/example/ApplicationTest.kt:16]: This test would fail. Can we either remove it or add another valid test?

**Review by vnikolova — CHANGES_REQUESTED** [review]
Nice example! Just a few details that need updating :)

**Review by vnikolova — APPROVED** [review]
Looks good!

**Review by e5l — APPROVED** [review]
lgtm

**Pull request merged**